com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CheckServicesToDeleteStep.java

```diff
@@ -0,0 +1,24 @@
+package com.sap.cloud.lm.sl.cf.process.steps;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+
+import com.sap.cloud.lm.sl.cf.client.lib.domain.CloudServiceExtended;
+import com.sap.cloud.lm.sl.cf.client.lib.domain.ImmutableCloudServiceExtended;
+
+@Component("checkServicesToDeleteStep")
+public class CheckServicesToDeleteStep extends CheckForOperationsInProgressStep {
+
+    @Override
+    protected List<CloudServiceExtended> getServices(ExecutionWrapper execution) {
+        List<CloudServiceExtended> services = new ArrayList<>();
+        StepsUtil.getServicesToDelete(execution.getContext())
+            .forEach(serviceName -> services.add(ImmutableCloudServiceExtended.builder()
+                .name(serviceName)
+                .build()));
```

Comment:
Also, if you really need to copy it, you could refactor it like this:
`List<CloudServiceExtended> servicesToDelete = StepsUtil.getServicesToDelete(execution.getContext());
 return servicesToDelete.stream().map(serviceToDelete -> copy(serviceToDelete)).collect(Collectors.toList();`
Where the copy is extract method containing: 
`ImmutableCloudServiceExtended.builder()
.name(serviceName)
                 .build()));`

Reply:
+1